### PR TITLE
[EDU-3160] feat: secure edge functions

### PR DESCRIPTION
**Pull request by gabriel-azion**
### Related issue:

https://aziontech.atlassian.net/browse/EDU-3160

### Changes

For example:
- Add automate secure folder
- Add automate with edge functions
- Implement small improvements on other 2 related files

### CODE REVIEW

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+description: Through Edge Functions you can write your own security source code in JavaScript and deploy it to run at the edge of the network.
+meta_tags: edge, security, javascript
+namespace: documentation_secure_automate_functions
+menu_namespace: secureMenu
+permalink: /documentation/products/secure/automate/functions/
+---
+
+import Button from '~/components/Button.astro'
+
+
```
Comment (hannah-sk) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:12]:
```suggestion
```

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+---
+
+## Develop edge functions for protection
+
+The Azion Edge Functions for the firewall run in the request phase, and respect the following process: 
+
+1. The rules configured on the [Edge Firewall Rules Engine](/en/documentation/products/edge-firewall/rules-engine/) for the function to run are triggered
+
+2. Azion Edge Runtime processes the function, returning an outcome
+
```
Comment (hannah-sk) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:34]:
```suggestion
1. The rules configured on the [Edge Firewall Rules Engine](/en/documentation/products/edge-firewall/rules-engine/) for the function to run are triggered.
2. Azion Edge Runtime processes the function, returning an outcome.
```

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+3. Edge Firewall Rules Engine resumes the processing, based on the outcome, from the point the behavior was triggered.
+
+
+<Button href="/en/documentation/products/edge-firewall/edge-functions/firewall/" text="go to edge functions for edge firewall reference" variant="secondary" target="_blank"></Button>
+
+---
+
+## Secure with APIs
+
+[Azion Edge Runtime](/en/documentation/products/edge-application/edge-functions/runtime/overview) supports a list of APIs that allow manipulation of [metadata](/en/documentation/products/secure/automate/functions/#metadata) and the use of [network lists]/en/documentation/products/secure/automate/functions/#metadata) to help the logic implementation.
```
Comment (hannah-sk) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:44]:
```suggestion
[Azion Edge Runtime](/en/documentation/products/edge-application/edge-functions/runtime/overview) supports a list of APIs that allow manipulation of [metadata](/en/documentation/products/secure/automate/functions/#metadata) and the use of [network lists](/en/documentation/products/edge-application/edge-functions/runtime/api-reference/network-list/) to help the logic implementation.
```

Comment (hannah-sk) [reply src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:44]: check 2nd link

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+The Metadata API enables access to a set of metadata that can be manipulated to:
+
+- Filter and manage access to your application
+- Apply specific logic in different scenarios based on GeoIP
+
+<Button href="/en/documentation/products/edge-application/edge-functions/runtime/api-reference/metadata/" text="go to metadata api reference" variant="secondary" target="_blank"></Button>
+
+### Network list
+
+Azion Network List Interface allows matching an IP address against a specific network list. If the IP address informed is within the network list, it returns true and the desired implementation logic can go on based on that information.
```
Comment (hannah-sk) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:60]:
```suggestion
Azion Network List Interface allows matching an IP address against a specific network list. If the IP address informed is within the network list, it returns `true` and the desired implementation logic can go on based on that information.
```

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+---
+title: Automate security with Azion Edge Functions
+description: Through Edge Functions you can write your own security source code in JavaScript and deploy it to run at the edge of the network.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:3]:
```suggestion
description: Through Edge Functions, you can write your own security source code in JavaScript and deploy it to run at the edge of the network.
```

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx (hunk 3 above)
Comment (gabriel-azion) [reply src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:40]:
```suggestion
[Azion Edge Runtime](/en/documentation/products/edge-application/edge-functions/runtime/overview) supports a list of APIs that allow manipulation of [metadata](/en/documentation/products/secure/automate/functions/#metadata) and the use of [network lists](/en/documentation/products/secure/automate/functions/##network-list) to help the logic implementation.
```

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+## Edge Functions and Edge Firewall
+
+Edge Functions are functions that run on the Azion Edge Computing Platform with low latency, bringing the operational power closer to the end user.
+
+Through Edge Functions you can write your own security source code in **JavaScript** and deploy it to run at the edge of the network, allowing you to:
+
+- Boost your protection
+- Have more dynamism
+- Apply the logic that your business requires
+- Use APIs that provide request and response header manipulation
```
Comment (bru-andrade) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:21]:
```suggestion
- Use APIs to manipulate request and response headers
```

**Review by bru-andrade — APPROVED** [review]

File: src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+permalink: /documentation/products/secure/automate/functions/
+---
+
+import Button from '~/components/Button.astro'
+
+## Edge Functions and Edge Firewall
+
+Edge Functions are functions that run on the Azion Edge Computing Platform with low latency, bringing the operational power closer to the end user.
+
+Through Edge Functions you can write your own security source code in **JavaScript** and deploy it to run at the edge of the network, allowing you to:
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/secure-journey/automate/edge-functions-apis.mdx:16]:
```suggestion
Through Edge Functions, you can write your own security source code in **JavaScript** and deploy it to run at the edge of the network, allowing you to:
```

**Review by hannah-sk — APPROVED** [review]
✌️

**Pull request merged**